### PR TITLE
test: Code Coverage: Add Edit Mileage: #9: Part 1

### PR DESCRIPTION
**Pull request by jayfyle**
### <samp>🤖 Generated by Copilot at 421a131</samp>

This pull request enhances the add/edit mileage page component and its test files. It adds mock data for different scenarios of mileage rates, org settings, and custom fields. It also refactors the component to use a new service for back button handling and removes some unused code.

### <samp>🤖 Generated by Copilot at 421a131</samp>

> _Sing, O Muse, of the skillful coder who devised_
> _Many mock data and constants for the `mileageRateOptions`_
> _And tested the `add/edit mileage page` with various scenarios_
> _Of autofill, projects, custom fields, and cost centers._

### <samp>🤖 Generated by Copilot at 421a131</samp>

*  Add mock data for mileage rates, org settings, org user settings, and unflattened transactions ([link](https://github.com/fylein/fyle-mobile-app/pull/2342/files?diff=unified&w=0#diff-133e43a25a6853af22f2d26760576e6675a1c86b504aa87b9f12b3ddad8d7e0eR322-R366), [link](https://github.com/fylein/fyle-mobile-app/pull/2342/files?diff=unified&w=0#diff-1cac010b51c1a7c2fbdf9bf224a7e267779365abe08b0f16df1ef46e41483951R1316-R1332), [link](https://github.com/fylein/fyle-mobile-app/pull/2342/files?diff=unified&w=0#diff-de467ea1b214e32a89452896f193a5769a02286b73795b8723305087a9ad59c9R673-R677), [link](https://github.com/fylein/fyle-mobile-app/pull/2342/files?diff=unified&w=0#diff-21ae928ce640ad62cfada5fee7bf0bbc490300ab04a25f8309a9878e21b2f1cfR3520-R3556))
*  Add imports for accessing and manipulating DOM elements, providing mock data, and using helper functions and enums in the add/edit mileage page spec file ([link](https://github.com/fylein/fyle-mobile-app/pull/2342/files?diff=unified&w=0#diff-f6fc308cee7591078e8b657e64b7b505b2b90996316c5ee098bc4c557c81eddbL5-R5), [link](https://github.com/fylein/fyle-mobile-app/pull/2342/files?diff=unified&w=0#diff-f6fc308cee7591078e8b657e64b7b505b2b90996316c5ee098bc4c557c81eddbL46-R47), [link](https://github.com/fylein/fyle-mobile-app/pull/2342/files?diff=unified&w=0#diff-f6fc308cee7591078e8b657e64b7b505b2b90996316c5ee098bc4c557c81eddbL60-R68), [link](https://github.com/fylein/fyle-mobile-app/pull/2342/files?diff=unified&w=0#diff-f6fc308cee7591078e8b657e64b7b505b2b90996316c5ee098bc4c557c81eddbL71-R88))
*  Add a spy object for the `PlatformHandlerService` service and assign it to the `TestBed` injected instance in the add/edit mileage page spec file ([link](https://github.com/fylein/fyle-mobile-app/pull/2342/files?diff=unified&w=0#diff-f6fc308cee7591078e8b657e64b7b505b2b90996316c5ee098bc4c557c81eddbR141), [link](https://github.com/fylein/fyle-mobile-app/pull/2342/files?diff=unified&w=0#diff-f6fc308cee7591078e8b657e64b7b505b2b90996316c5ee098bc4c557c81eddbR201))
*  Add test cases for the `ionViewWillEnter`, `getMileageRatesOptions`, and `initSubjectObservables` methods of the add/edit mileage page component in the spec file ([link](https://github.com/fylein/fyle-mobile-app/pull/2342/files?diff=unified&w=0#diff-f6fc308cee7591078e8b657e64b7b505b2b90996316c5ee098bc4c557c81eddbL411-R680))
*  Add a spy object for the `PlatformHandlerService` service and a provider that uses it in the add/edit mileage page setup spec file ([link](https://github.com/fylein/fyle-mobile-app/pull/2342/files?diff=unified&w=0#diff-df40b6e711e89d5e80e6f3867394ee96bbff1eeae11eef42db09ce0b3f1821fbR228-R229), [link](https://github.com/fylein/fyle-mobile-app/pull/2342/files?diff=unified&w=0#diff-df40b6e711e89d5e80e6f3867394ee96bbff1eeae11eef42db09ce0b3f1821fbR438-R441))
*  Remove the import of `Platform` and add the import of `PlatformHandlerService` in the add/edit mileage page component file ([link](https://github.com/fylein/fyle-mobile-app/pull/2342/files?diff=unified&w=0#diff-04c216da6200af711aa0f53bcd7e253086762b5d123b0729fac3b1862ec5fc3eL8-R8), [link](https://github.com/fylein/fyle-mobile-app/pull/2342/files?diff=unified&w=0#diff-04c216da6200af711aa0f53bcd7e253086762b5d123b0729fac3b1862ec5fc3eR104))
*  Remove the `ViewChild` decorators for unused elements and add a type annotation for the `formContainer` element in the add/edit mileage page component file ([link](https://github.com/fylein/fyle-mobile-app/pull/2342/files?diff=unified&w=0#diff-04c216da6200af711aa0f53bcd7e253086762b5d123b0729fac3b1862ec5fc3eL134-R136))
*  Remove the `platform` parameter and add the `platformHandlerService` parameter in the constructor of the add/edit mileage page component ([link](https://github.com/fylein/fyle-mobile-app/pull/2342/files?diff=unified&w=0#diff-04c216da6200af711aa0f53bcd7e253086762b5d123b0729fac3b1862ec5fc3eL324-R321))
*  Replace the direct subscription to the `platform.backButton` observable with a call to the `platformHandlerService.registerBackButtonAction` method in the `ngOnInit` method of the add/edit mileage page component ([link](https://github.com/fylein/fyle-mobile-app/pull/2342/files?diff=unified&w=0#diff-04c216da6200af711aa0f53bcd7e253086762b5d123b0729fac3b1862ec5fc3eL948-R949))
*  Remove the type assertion for the `formContainer` element in the `showFormValidationErrors` method of the add/edit mileage page component ([link](https://github.com/fylein/fyle-mobile-app/pull/2342/files?diff=unified&w=0#diff-04c216da6200af711aa0f53bcd7e253086762b5d123b0729fac3b1862ec5fc3eL1783-R1778))
*  Remove the unused `scrollCommentsIntoView` method from the add/edit mileage page component ([link](https://github.com/fylein/fyle-mobile-app/pull/2342/files?diff=unified&w=0#diff-04c216da6200af711aa0f53bcd7e253086762b5d123b0729fac3b1862ec5fc3eL2645-L2657))

## Clickup
https://app.clickup.com/t/85ztjmtg0

## Code Coverage
`97.47% Statements 812/833`
`91.8% Branches 650/708`
`97.25% Functions 283/291`
`97.48% Lines 774/794`

## UI Preview
Please add screenshots for UI changes

### CODE REVIEW

File: src/app/fyle/add-edit-mileage/add-edit-mileage-5.spec.ts [hunk 1]
```diff
@@ -2,7 +2,7 @@ import { TitleCasePipe } from '@angular/common';
 import { ComponentFixture, fakeAsync, tick } from '@angular/core/testing';
 import { FormBuilder, Validators, FormArray } from '@angular/forms';
 import { MatSnackBar } from '@angular/material/snack-bar';
-import { DomSanitizer } from '@angular/platform-browser';
+import { By, DomSanitizer } from '@angular/platform-browser';
```
Comment (Dimple16) [inline src/app/fyle/add-edit-mileage/add-edit-mileage-5.spec.ts:5]: are we using this? looks like it is imported by mistake, please check and remove if unused

File: src/app/fyle/add-edit-mileage/add-edit-mileage-5.spec.ts [hunk 2]
```diff
@@ -408,6 +425,258 @@ export function TestCases5(getTestBed) {
         expect(component.getMileageByVehicleType).toHaveBeenCalledOnceWith(unfilteredMileageRatesData, null);
         expect(mileageRatesService.getReadableRate).toHaveBeenCalledOnceWith(null, 'INR', null);
       }));
+
+      it('should setup class variables with autofill enabled and no recent values available', fakeAsync(() => {
```
Comment (Dimple16) [inline src/app/fyle/add-edit-mileage/add-edit-mileage-5.spec.ts:429]: can we add the common code in `ionviewWillEnter` related test cases together in a method/before each if that works? 
there's lot of repetition and test case is getting very long, difficult to read

**Review by Dimple16 — CHANGES_REQUESTED** [review]
added comments

**Review by OmkarJ13 — COMMENTED** [review]
There are few merge conflicts, can you pls re-assign once the conflicts are fixed, will review afterwards

**Review by Dimple16 — APPROVED** [review]

**Review by OmkarJ13 — APPROVED** [review]

**Review by Chethan-Fyle — APPROVED** [review]

**Pull request merged**